[Docs] Catalog promotion label customization (#13386)
Sylius/Sylius: docs/cookbook/promotions/customize-catalog-promotion-labels.rst

How to customize catalog promotion labels?
==========================================

> To customize catalog promotion labels only what need to do is a overwrite proper template

Reviewer: ```suggestion
To customize catalog promotion labels only what need to do is overwrite the proper template
```

How to customize catalog promotion labels?
==========================================

To customize catalog promotion labels only what need to do is a overwrite proper template

.. tip::

    Learn more about template customization :doc:`here </customization/template>`.

> Changing styles of Catalog Promotion labels.
> --------------------------------------------

Reviewer 2: ```suggestion
Change the catalog promotion labels' styles
-------------------------------------------
```

> In default Sylius we show catalog promotion labels in 2 places: on product index and product show page. So we need to customize  2 related templates:

Reviewer 2: ```suggestion
By default Sylius displays catalog promotion labels in two locations: on the products' index on product's card and also on the product show page. That's why you'll need to customize two related templates:
```

> For product index, we will customize template `@SyliusShopBundle/Product/Box/_content.html.twig`

Reviewer 2: ```suggestion
For product index, we will customize the ``@SyliusShopBundle/Product/Box/_content.html.twig`` template
```

> by creating a file ``templates/bundles/SyliusShopBundle/Product/Box/_content.html.twig`` with content:

Reviewer 2: ```suggestion
by overriding it with your own ``templates/bundles/SyliusShopBundle/Product/Box/_content.html.twig`` that will have such content:
```

                {% set appliedPromotions = (product|sylius_resolve_variant).getAppliedPromotionsForChannel(sylius.channel) %}
                {% set price = money.calculatePrice(product|sylius_resolve_variant) %}
                {% set originalPrice = money.calculateOriginalPrice(product|sylius_resolve_variant) %}

                {% for promotion in appliedPromotions %}
                {#  here you need to add your classes of styles #}
                    <a class="ui red big label" {{ sylius_test_html_attribute('product-promotion-label') }}>{{ promotion.label}}</a>
                {% endfor %}

                {% if price != originalPrice %}
                    <div class="sylius-product-original-price" {{ sylius_test_html_attribute('product-original-price') }}><del>{{ originalPrice }}</del></div>
                {% endif %}
                <div class="sylius-product-price" {{ sylius_test_html_attribute('product-price') }}>{{ price }}</div>
            {% endif %}
        </div>
    </div>

After changes:

.. image:: ../../_images/cookbook/customization-catalog-promotion/catalog_promotion_label_product_index.png
    :align: center

> For product show page we will customize template `@SyliusShopBundle/Product/Show/_price.html.twig`
> by creating a file ``templates/bundles/SyliusShopBundle/Product/Show/_price.html.twig`` with content:

Reviewer 2: ```suggestion
For product show page you'll need to override the ``@SyliusShopBundle/Product/Show/_price.html.twig`` template with ``templates/bundles/SyliusShopBundle/Product/Show/_price.html.twig`` looking like that:
```

        {% if hasDiscount %}
            <del>{{ money.calculateOriginalPrice(variant) }}</del>
        {% endif %}
    </span>

    <div id="appliedPromotions" data-applied-promotions-locale="{{ sylius.localeCode }}">
        {% for appliedPromotion in channelPricing.appliedPromotions %}
            {#  here you need to add your classes of styles #}
            <div class="ui red big label promotion_label" style="margin: 1rem 0;">
                <div class="row ui small sylius_catalog_promotion">
                    {{ appliedPromotion.label }}{% if appliedPromotion.description %} - {{ appliedPromotion.description }}{% endif %}
                </div>
            </div>
        {% endfor %}
    </div>

    <span class="ui huge header" id="product-price" {{ sylius_test_html_attribute('product-price', money.calculatePrice(variant)) }}>
        {{ money.calculatePrice(variant) }}
    </span>

After changes:

.. image:: ../../_images/cookbook/customization-catalog-promotion/catalog_promotion_label_product_show.png
    :align: center

> 

Reviewer 2: ```suggestion

Of course, you can modify the styles how you need, also in your stylesheet.
```

> Display catalog promotion labels on photo instead of under it

Reviewer 2: ```suggestion
Display catalog promotion labels on the product image
```

-------------------------------------------------------------

> For changing place of catalog promotion label on product index, we need to overwrite 2 templates:

Reviewer 2: ```suggestion
In order to change the location of catalog promotion label on the product's card, we need to overwrite two templates. We will remove the label from underneath the photo in product card template and then add it on the product's image.
```

> template `@SyliusShopBundle/Product/Box/_content.html.twig`
> by creating a file ``templates/bundles/SyliusShopBundle/Product/Box/_content.html.twig`` with content:

Reviewer 2: ```suggestion
Override the ``@SyliusShopBundle/Product/Box/_content.html.twig`` template with
``templates/bundles/SyliusShopBundle/Product/Box/_content.html.twig``:
```

                    </div>
                </div>
            </div>
            {% include '@SyliusShop/Product/_mainImage.html.twig' with {'product': product} %}
        </a>
        <div class="content" {{ sylius_test_html_attribute('product-content') }}>
            <a href="{{ path('sylius_shop_product_show', {'slug': product.slug, '_locale': product.translation.locale}) }}" class="header sylius-product-name" {{ sylius_test_html_attribute('product-name', product.name) }}>{{ product.name }}</a>

            {% if not product.enabledVariants.empty() %}
                {% set price = money.calculatePrice(product|sylius_resolve_variant) %}
                {% set originalPrice = money.calculateOriginalPrice(product|sylius_resolve_variant) %}
                {% if price != originalPrice %}
                    <div class="sylius-product-original-price" {{ sylius_test_html_attribute('product-original-price') }}><del>{{ originalPrice }}</del></div>
                {% endif %}
                <div class="sylius-product-price" {{ sylius_test_html_attribute('product-price') }}>{{ price }}</div>
            {% endif %}
        </div>
    </div>

> And template `@SyliusShopBundle/Product/_mainImage.html.twig` by creating a file ``templates/bundles/SyliusShopBundle/Product/_mainImage.html.twig`` with content:

Reviewer 2: ```suggestion
And the ``@SyliusShopBundle/Product/_mainImage.html.twig`` with ``templates/bundles/SyliusShopBundle/Product/_mainImage.html.twig``:
```

    {% else %}
        {% set path = '//placehold.it/200x200' %}
    {% endif %}

    {% set variant = product|sylius_resolve_variant %}
    {% set hasDiscount = variant|sylius_has_discount({'channel': sylius.channel}) %}
    {% set channelPricing = variant.getChannelPricingForChannel(sylius.channel) %}

    <div style="position: relative;">
        <img src="{{ path }}" {{ sylius_test_html_attribute('main-image') }} alt="{{ product.name }}" class="ui bordered image" />
        <div id="appliedPromotions" style="position: absolute; right: 5px; top: 5px" data-applied-promotions-locale="{{ sylius.localeCode }}">
            {% for appliedPromotion in channelPricing.appliedPromotions %}
                <div class="ui blue label promotion_label" style="margin: 1rem 0;">
                    <div class="row ui small sylius_catalog_promotion">
                        {{ appliedPromotion.name }}
                    </div>
                </div>
            {% endfor %}
        </div>
    </div>

After changes:

.. image:: ../../_images/cookbook/customization-catalog-promotion/catalog_promotion_repositioned_label.png

>     :align: center

Reviewer 2: ```suggestion
    :align: center
    
    Well done! Now you can do anything you want with the catalog promotion labels.
```

🎉
